app/routes/login-callback/route.tsx

```diff
@@ -0,0 +1,37 @@
+import { LoaderFunctionArgs, redirect } from '@remix-run/node';
+import { commitSession, getSession, initializeEcomApiForRequest } from '~/lib/ecom/session';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+    const session = await getSession(request.headers.get('Cookie'));
+    const storedOauthData = session.get('oAuthData');
+
+    if (storedOauthData === undefined) {
+        return redirect('/login');
+    }
+
+    const api = await initializeEcomApiForRequest(request);
+    const wixClient = api.getWixClient();
+
+    const returnedOAuthData = wixClient.auth.parseFromUrl(request.url, 'query');
+    if (returnedOAuthData.error) {
+        throw new Error(`Error: ${returnedOAuthData.errorDescription}`);
```

Comment:
will the root error boundary catch this and show nice error UI?

Reply:
no, it will look like this:
<img width="949" alt="image" src="https://github.com/user-attachments/assets/697613fa-30d4-44bc-a111-cc2c2aca0dc3">

Reply:
no, it will look like this:
![image](https://github.com/user-attachments/assets/2a0bea2a-e5ad-47f8-82d8-480507a9a1bf)

Reply:
not cool

Reply:
I can see two options to make it better:
1. create a component and error boundary for this route (not desired, cause that page will be visible in codux and basically it is not clear what to there - spinner?)
2. rediret to some `/error` route, but this is also not good, we were trying to omit separate error page

Reply:
might look like this, but in this case, page appear in codux:
![image](https://github.com/user-attachments/assets/b806b44d-fdba-4919-8231-f3c1572cc3ba)